### PR TITLE
Support usage in Pipeline Plugin Scripts

### PR DESCRIPTION
**Pull request by djaara**
This PR allows this plugin to be used inside of the Pipeline Plugin by implementing the `SimpleBuildStep` interface.

See `readme.md` for an example of how it would be used.

Based on PR #112, but removed all whitespace changes and all Java 7 dependencies (multi catch, try-with-resources) as Java 6 is default with `org.jenkins-ci.plugins:plugin:1.577`

### CODE REVIEW

**Comment by scaytrase** [comment]
Brief look says that it's much better, than #112. I'll check in depth it as soon as I can. Thanks.

File: pom.xml [hunk 1]
```diff
@@ -75,6 +75,11 @@
 
 	<dependencies>
 		<dependency>
+			<groupId>com.google.guava</groupId>
```
Comment (scaytrase) [inline pom.xml]: What's the purpose of this? Could not see any usage of `com.google` namespace

**Comment by scaytrase** [comment]
Looks good, minor question on a dependency above. I'll try to test it by myselft before merge

File: pom.xml (hunk 1 above)
Comment (djaara) [inline pom.xml]: Without this dependency there are some failed tests with following exception:

```
java.lang.NoClassDefFoundError: com/google/common/util/concurrent/ForwardingExecutorService

	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.powermock.core.classloader.MockClassLoader.loadModifiedClass(MockClassLoader.java:178)
	at org.powermock.core.classloader.DeferSupportingClassLoader.loadClass(DeferSupportingClassLoader.java:68)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at hudson.FilePath.<clinit>(FilePath.java:2568)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at javassist.runtime.Desc.getClassObject(Desc.java:43)
	at javassist.runtime.Desc.getClassType(Desc.java:152)
	at javassist.runtime.Desc.getType(Desc.java:122)
	at javassist.runtime.Desc.getType(Desc.java:78)
	at org.jenkinsci.plugins.stashNotifier.StashNotifierTest.test_perform_simplebuildstep(StashNotifierTest.java:265)
	at org.jenkinsci.plugins.stashNotifier.StashNotifierTest.test_perform_simple_build_step_success(StashNotifierTest.java:368)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.internal.runners.TestMethod.invoke(TestMethod.java:68)
	at org.powermock.modules.junit4.internal.impl.PowerMockJUnit44RunnerDelegateImpl$PowerMockJUnit44MethodRunner.runTestMethod(PowerMockJUnit44RunnerDelegateImpl.java:310)
	at org.junit.internal.runners.MethodRoadie$2.run(MethodRoadie.java:89)
	at org.junit.internal.runners.MethodRoadie.runBeforesThenTestThenAfters(MethodRoadie.java:97)
	at org.powermock.modules.junit4.internal.impl.PowerMockJUnit44RunnerDelegateImpl$PowerMockJUnit44MethodRunner.executeTest(PowerMockJUnit44RunnerDelegateImpl.java:294)
	at org.powermock.modules.junit4.internal.impl.PowerMockJUnit47RunnerDelegateImpl$PowerMockJUnit47MethodRunner.executeTestInSuper(PowerMockJUnit47RunnerDelegateImpl.java:127)
	at org.powermock.modules.junit4.internal.impl.PowerMockJUnit47RunnerDelegateImpl$PowerMockJUnit47MethodRunner.executeTest(PowerMockJUnit47RunnerDelegateImpl.java:82)
	at org.powermock.modules.junit4.internal.impl.PowerMockJUnit44RunnerDelegateImpl$PowerMockJUnit44MethodRunner.runBeforesThenTestThenAfters(PowerMockJUnit44RunnerDelegateImpl.java:282)
	at org.junit.internal.runners.MethodRoadie.runTest(MethodRoadie.java:87)
	at org.junit.internal.runners.MethodRoadie.run(MethodRoadie.java:50)
	at org.powermock.modules.junit4.internal.impl.PowerMockJUnit44RunnerDelegateImpl.invokeTestMethod(PowerMockJUnit44RunnerDelegateImpl.java:207)
	at org.powermock.modules.junit4.internal.impl.PowerMockJUnit44RunnerDelegateImpl.runMethods(PowerMockJUnit44RunnerDelegateImpl.java:146)
	at org.powermock.modules.junit4.internal.impl.PowerMockJUnit44RunnerDelegateImpl$1.run(PowerMockJUnit44RunnerDelegateImpl.java:120)
	at org.junit.internal.runners.ClassRoadie.runUnprotected(ClassRoadie.java:34)
	at org.junit.internal.runners.ClassRoadie.runProtected(ClassRoadie.java:44)
	at org.powermock.modules.junit4.internal.impl.PowerMockJUnit44RunnerDelegateImpl.run(PowerMockJUnit44RunnerDelegateImpl.java:122)
	at org.powermock.modules.junit4.common.internal.impl.JUnit4TestSuiteChunkerImpl.run(JUnit4TestSuiteChunkerImpl.java:106)
	at org.powermock.modules.junit4.common.internal.impl.AbstractCommonPowerMockRunner.run(AbstractCommonPowerMockRunner.java:53)
	at org.powermock.modules.junit4.PowerMockRunner.run(PowerMockRunner.java:59)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:117)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:42)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:253)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:84)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: java.lang.ClassNotFoundException: com.google.common.util.concurrent.ForwardingExecutorService
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 77 more
```

Comment (scaytrase) [inline pom.xml]: Okay, kinda dev-dependencies, thanks for clarifying

**Comment by scaytrase** [comment]
Test OK, basic cases works well with pipeline and still with free configuration job.

**Comment by scaytrase** [comment]
Thanks @ckuhn, thanks @djaara for getting things right. I'll try to release that as soon as possible

**Comment by mrooney** [comment]
Thanks for the awesome work @djaara ! One issue I noticed is that when used in a Pipeline right after checking out the scm, the normal data doesn't seem to get sent to Bitbucket. See the attached screenshot where the first build is from a job hitting via a Pipeline script and the second is from checking the box in the config UI and it gets the job name and Jenkins URL. However, when I run StashNotifier again at the end of the job, then I end up with three builds, one with the correct name/description and one without, both pointing at the same job. Should I file a bug report for this?
screen shot 2016-11-15 at 6 36 58 pm

**Comment by scaytrase** [comment]
Not sure if it is a misconfiguration or a bug, but you are welcome to file an issue with more data to inspect.

**Comment by mrooney** [comment]
I see, the Bitbucket Branch Source plugin ALSO automatically sends these, sorry for the confusion :)

**Pull request merged**